### ISSUE TITLE
Support RFC 7529: non-gregorian calendars and leap-month handling

### ISSUE THREAD

**Issue by mainrs** [fmeringdal/rust-rrule#95]
https://datatracker.ietf.org/doc/rfc7529/

This is useful when working with repeating events from lunar calendars for example. The leap month proposal is also pretty nice when events fall onto the 29th February. As of know, I have to manually write exceptions for these events in my application.

**Comment by Lazare** [comment]
Even if only the GREGORIAN calendar is supported, the SKIP=(BACKWARD,FORWARD,OMIT) parameter introduced in RFC 7529 is very useful.
For example, if you set a monthly recurrence, and it starts on January 30th then the next iteration will be March 30th. It's very counterintuitive for most users, as they probably want it to be February the 28th (or 29th on leap years)

**Comment by mdecimus** [comment]
> Even if only the GREGORIAN calendar is supported, the SKIP=(BACKWARD,FORWARD,OMIT) parameter introduced in RFC 7529 is very useful.

I agree! Do you know any calendar clients that generate rules containing `SKIP`?

**Comment by mainrs** [comment]
> I agree! Do you know any calendar clients that generate rules containing SKIP?

I cannot export the event, but Proton Calendar does indeed skip it. If I create a recurring event on January 30th and tell it to repeat every month, the next event will be March 30th.

**Comment by Lazare** [comment]
After searching for several implementations in different languages, most don’t support the RSCALE extension. Some implementations perform an automatic backward skip (https://balsoftware.net/rrule/) and seem to calculate the next iterations incorrectly (March 28 instead of March 30).

My use is for services providers to define subscription billing rules for their services. For example, they can either have monthly billing at a predefined day of month (1st for example) or have sliding month (ie: if someone subscribe on the 23rd they will be billed on the 23rd of next month). They won't be happy if someone subscribing on the 31st of a month is billed only 7 times a year.
I'd like to use a standard solution and not invent my own.
